swan-lake/resources/featured-scenarios/create-your-first-connector-with-ballerina.md

```diff
@@ -0,0 +1,166 @@
+
+2. Clone your newly created empty repository to your local machine:
+   ```
+   git clone https://github.com/<your-username>/<connector-repo-name>.git
+   cd <connector-repo-name>
+   ```
+
+3. Visit the [Ballerina generated connector template on GitHub](https://github.com/ballerina-platform/ballerina-library/tree/main/library-templates/generated-connector-template/files) and copy the entire project structure and content to your local repository folder, making sure to include all files and directories.
+
+4. Your local project structure should now look similar to this:
+   ```
+   module-ballerinax-myconnector/
+   ├── .github/
+   ├── ballerina/
+   │   ├── Ballerina.toml
+   │   ├── Module.md
+   │   ├── Package.md
+   │   ├── build.gradle
+   │   └── client.bal
+   ├── build-config/
+   ├── docs/
+   │   └── spec/
+   │       └── sanitations.md
+   ├── examples/
+   │   ├── README.md
+   │   ├── build.gradle
+   │   └── build.sh
+   ├── gradle/
+   ├── .gitignore
+   ├── LICENSE
+   ├── README.md
+   ├── build.gradle
+   ├── gradle.properties
+   ├── gradlew
+   ├── gradlew.bat
+   └── settings.gradle
+   ```
+
+> **Tip**: The template includes placeholders for various fields. To update them with your connector-specific metadata, use the provided [Ballerina script](https://github.com/ballerina-platform/ballerina-library/blob/main/library-templates/generated-connector-template/scripts/replace_placeholders.bal).
+
+Detailed information on the Ballerina connector structure can be found in the [Ballerina module contribution guide](https://github.com/ballerina-platform/ballerina-library/blob/main/docs/adding-a-new-ballerina-module.md#directory-structure).
+
+## Step 2: Prepare the OpenAPI Specification
+
+1. Find the OpenAPI specification for the API you want to create a connector for. This is usually available in the API documentation.
+   Example: For Twitter, you can get the latest API specification from the [Twitter OpenAPI endpoint](https://api.twitter.com/2/openapi.json).
+
+2. Save the file as `spec.yaml` (or `spec.json`) in the `docs/spec` directory of your project.
+
+3. You may need to sanitize the OpenAPI specification to ensure compatibility with the Ballerina OpenAPI tool. This process may involve:
+   - Renaming schema names to comply with Ballerina naming conventions.
+   - Adding, removing, or modifying security schemes to customize authentication options.
+
+## Step 3: Generate the Ballerina client code
+
+With your OpenAPI spec ready, use the [Ballerina OpenAPI tool](https://ballerina.io/learn/openapi-tool/) to generate the connector code.
+
+1. In your terminal, run the following command from the project root:
+   ```
+   bal openapi -i path/to/spec --mode client -o ballerina
+   ```
+
+   This will generate the ballerina source code of the connector client in your `ballerina/` directory.
```

Comment:
connector client or client connector?

Reply:
+1 for client connector

Reply:
Updated in https://github.com/ballerina-platform/ballerina-dev-website/pull/9434/commits/25444cee3c67cbe26565114aff1190140114d44b